horovod/runner/common/util/safe_shell_exec.py

```diff
@@ -78,9 +78,9 @@ def terminate_executor_shell_and_children(pid):
             pass
 
 
-def forward_stream(src_stream, dst_stream, prefix, index):
+def forward_stream(src_stream, dst_stream, prefix, index, hide_timestamp):
```

Comment:
I wouldn't call this `hide_timestamp` but `with_timestamp=True`

Reply:
Good point.  Looking into this further, it looks like we have an inconsistency here between how `horovodrun` handles MPI vs Gloo.  With MPI, we need to set `--timestamp-output` to get this behavior, which we don't do by default.  So I think the best thing to do is change the default to Gloo to also hide the timestamp, then only enable (for both) when the user passes in `--timestamp-output`.  What do you think?

Reply:
I am happy with this being opt-in behaviour, for MPI and Gloo. Still, the `horovodrun` command line argument does not need to be called `--timestamp-output` just because we forward it to that flag when using MPI. Something like `--log-remote-shell-with-timestamp` would be meaningful.

Reply:
What about `--prepend-output-with-timestamp` / `-prepend-timestamp`?  I think we should avoid "log" here since this applies to all stdout / stderr, not only logs (which have a different meaning for other args).

Reply:
What about `--prefix-output-with-timestamp`? Otherwise `--prepend-output-with-timestamp` is fine.

Reply:
`prefix` is good, too.  Wasn't sure which to go with, so sounds like prefix it is.